### PR TITLE
snmp-ups: Restore legacy Eaton ePDU switchability info

### PR DESCRIPTION
**Pull request by aquette**
Use a hack to also have switchability for both the unit and its
outlets on legacy Eaton G2 ePDU

Signed-off-by: Arnaud Quette <ArnaudQuette@eaton.com>

### CODE REVIEW

**Comment by aquette** [comment]
Ref: IPMVAL-2314

File: drivers/eaton-pdu-marlin-mib.c [hunk 1]
```diff
@@ -801,10 +810,14 @@ static snmp_info_t eaton_marlin_mib[] = {
...
-	 * used to depict the overall outlets switchability of the unit */
+	 * used to depict the overall outlets switchability of the unit on G3 and newer ePDU*/
 	{ "outlet.switchable", ST_FLAG_STRING, SU_INFOSIZE,
 		".1.3.6.1.4.1.534.6.6.7.1.2.1.10.%i",
-		"no", 0, &marlin_unit_switchability_info[0] },
+		"no", SU_FLAG_STATIC | SU_FLAG_UNIQUE, &marlin_unit_switchability_info[0] },
+	/* Ugly hack for older G2 ePDU: check the first outlet to determine unit switchability */
+	{ "outlet.switchable", ST_FLAG_STRING, SU_INFOSIZE,
+		".1.3.6.1.4.1.534.6.6.7.6.6.1.3.%i.1",
+		"no", SU_FLAG_STATIC | SU_FLAG_UNIQUE | SU_OUTLET | SU_FLAG_OK | SU_TYPE_DAISY_1, &g2_unit_outlet_switchability_info[0] },
```
Comment (jimklimov) [inline drivers/eaton-pdu-marlin-mib.c:820]: Do I get this right that as the driver walks a device, it first hits the OID several lines above and sets the dstate value by marlin_unit_switchability_info mapping, then hits this OID too and re-sets the dstate by g2_unit_outlet_switchability_info?

Does not sound *too* ugly ;) Might even be upstreamable if that works too :)

File: scripts/DMF/dmfsnmp/eaton-pdu-marlin-mib.dmf [hunk 2]
```diff
@@ -1,41 +1,10 @@
 <?xml version="1.0" ?>
 <nut version="1.0.0" xmlns="http://www.networkupstools.org/dmf/snmp/snmp-ups">
-	<lookup name="marlin_threshold_voltage_alarms_info">
```
Comment (jimklimov) [inline scripts/DMF/dmfsnmp/eaton-pdu-marlin-mib.dmf:3]: Strange how the DMF's contents got shuffled :\ Can this be somehow build-locale dependent?

**Review by jimklimov — APPROVED** [review]

**Review by boricj — APPROVED** [review]
As with #119, seems safe enough for non-G2 ePDUs.

File: drivers/eaton-pdu-marlin-mib.c (hunk 1 above)
Comment (aquette) [reply drivers/eaton-pdu-marlin-mib.c:820]: not exactly for the behavior:
* on G2, the first outlet.switchable doesn't exist, so try the 2nd, which is the ugly hack: try on the first outlet to see if switchable
* on G3, the first is good, so flag unique does the rest

can be upstreamed indeed, though not enough time on my side as usual

File: scripts/DMF/dmfsnmp/eaton-pdu-marlin-mib.dmf (hunk 2 above)
Comment (aquette) [reply scripts/DMF/dmfsnmp/eaton-pdu-marlin-mib.dmf:3]: can't say. Another mystery of life ;)

**Comment by aquette** [comment]
btw, FYI tested on MA G2 and G3. now merging

**Pull request merged**